src/frontend/src/lib/derived/network.derived.ts

```diff
@@ -19,9 +20,19 @@ export const networkId: Readable<NetworkId> = derived(
 			: DEFAULT_NETWORK_ID
 );
 
+export const selectedNetwork: Readable<Network> = derived(
+	[networks, networkId],
+	([$networks, $networkId]) => $networks.find(({ id }) => id === $networkId) ?? DEFAULT_NETWORK
+);
+
 export const networkTokens: Readable<Token[]> = derived(
-	[tokens, networkId],
-	([$tokens, $networkId]) => $tokens.filter(({ network: { id } }) => id === $networkId)
+	[tokens, selectedNetwork],
+	([$tokens, $selectedNetwork]) => {
+		if (CHAIN_FUSION_NETWORKS_IDS.includes($selectedNetwork.id)) {
```

Comment:
agreed with the utils

for selectedNetwork: i didn't want to hard-code `'mainnet'` as comparison, but use the `selectedNetwork.env`, really just to avoid any hard-coded comparison, but even to be ready for a testnet in the future

Reply:
Ahhh now I get it, that's a good idea!